content/sensu-go/6.3/observability-pipeline/observe-schedule/rule-templates.md

```diff
@@ -0,0 +1,825 @@
+        default: non-zero
+        enum:
+          - non-zero
+          - warning
+          - critical
+          - unknown
+      threshold:
+        type: number
+        description: Numeric value that triggers an event when surpassed
+  eval: |
+    var statusMap = {
+      "non-zero": 1,
+      "warning": 1,
+      "critical": 2,
+    };
+    function main(args) {
+      var total = sensu.events.count();
+      var num = sensu.events.count(args.status);
+      if (num / total <= args.threshold) {
+        return;
+      }
+      return event.status = statusMap[args.status],
+      });
+    }
+{{< /code >}}
+
+{{< code json >}}
+{
+  "type": "RuleTemplate",
+  "api_version": "bsm/v1",
+  "metadata": {
+    "name": "status-threshold",
+    "namespace": "default"
+  },
+  "spec": {
+    "description": "Creates an event when the percentage of events with the given status exceed the given threshold",
+    "arguments": {
+      "required": [
+        "threshold"
+      ],
+      "properties": {
+        "status": {
+          "type": "string",
+          "default": "non-zero",
+          "enum": [
+            "non-zero",
+            "warning",
+            "critical",
+            "unknown"
+          ]
+        },
+        "threshold": {
+          "type": "number",
+          "description": "Numeric value that triggers an event when surpassed"
+        }
+      }
+    },
+    "eval": "var statusMap = {\n  \"non-zero\": 1,\n  \"warning\": 1,\n  \"critical\": 2,\n};\nfunction main(args) {\n  var total = sensu.events.count();\n  var num = sensu.events.count(args.status);\n  if (num / total <= args.threshold) {\n    return;\n  }\n  return event.status = statusMap[args.status],\n  });\n}"
+  }
+}
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+## Built-in rule template: Aggregate
+
+Sensu's business service monitoring (BSM) includes a built-in rule template, `aggregate`, that allows you to treat the results of multiple disparate check executions executed across multiple disparate systems as a single result (event).
+This built-in rule templates are ready to use with your service components.
+
+Reference the rule template name in the `rules.template` field and configure the arguments in the `rules.template.arguments` object in your service component resource definitions.
+
+### Aggregate rule template example
+
+Use the `aggregate` rule template for services that can be considered healthy as long as a minimum threshold is satisfied.
+For example, you might set the minimum threshold at 10 web servers with an OK status or 70% of processes running with an OK status.
+
+The `aggregate` rule template is useful in dynamic environments and environments with some tolerance for failure.
+
+To view the `aggregate` resource definition:
+
+{{< language-toggle >}}
+
+{{< code yml >}}
+sensuctl RuleTemplate info aggregate --format yaml
+{{< /code >}}
+
+{{< code json >}}
+sensuctl RuleTemplate info aggregate --format wrapped-json
+{{< /code >}}
+
```

Comment:
I don't know if these commands will actually work -- I have not been able to get sensuctl info to work with either rule templates or service components.

Reply:
Confirmed they do not work: https://sensu.slack.com/archives/C60EEQFH8/p1616508141272600